### ISSUE TITLE
Lua string with zero embedded

### ISSUE THREAD

**Issue by mbsteixeira** [surparallel/pelagia#31]
It is possible that the string of characters written in Lua may contain zeroes, in this case the use of strlen can generate errors. 
I noticed that in the source file plapi.c, there are a large number of this situation. How do I submit the changes I have made to correct this problem?

**Comment by gantleman** [comment]
Of course. I'll merge it after reading. Thank you for your support.

**Comment by mbsteixeira** [comment]
Great!
I have created the function static void* LPrepareGetDic(lua_State* L) and changed the int plg_LvmTableNext(void* plVMHandle, void* L, int i, char **k, size_t *klen, char **sv, size_t *len, double *dv) , including two parameters. 

If you identify any wrong implementation, please, notify me that I'll correct them.

[src_files.zip](https://github.com/surparallel/pelagia/files/5091828/src_files.zip)

**Comment by gantleman** [comment]
Very good! thank you!

my english is poor
The following text is also with the help of translation software, so the grammar will be a little strange.
Pelagia is part of my promotion of new parallel development technologies.
Pelagia will be permanently open source.
But I don't reject business plans.
I will develop non open source commercial software based on Pelagia.
Because I don't want to make Linux mistakes, because of excessive cleanliness, give the market to other businessmen for profit.
Please let me know if you agree to use your submitted code for commercialization and non open source.

________________________________
发件人: mbsteixeira <notifications@github.com>
发送时间: 2020年8月18日 17:26
收件人: surparallel/pelagia <pelagia@noreply.github.com>
抄送: gantleman <dds_sun@hotmail.com>; Comment <comment@noreply.github.com>
主题: Re: [surparallel/pelagia] Lua string with zero embedded (#31)


Great!
I have created the function static void* LPrepareGetDic(lua_State* L) and changed the int plg_LvmTableNext(void* plVMHandle, void* L, int i, char **k, size_t *klen, char **sv, size_t *len, double *dv) , including two parameters.

If you identify any wrong implementation, please, notify me that I'll correct them.

src_files.zip<https://github.com/surparallel/pelagia/files/5091828/src_files.zip>

―

**Comment by mbsteixeira** [comment]
I agree with you.  You can   use my  submitted code for commercialization and non open source products.
If you need help,  count on me.
I am not an english speaker, too., I am from Brazil.
